### ISSUE TITLE
.Net 8 Android App crashes only in release on physical device 

### ISSUE THREAD

**Issue by jonathanantoine** [dotnet/runtime#98232]
### Description

My app crashes when deployed on a physical device.

I have no  meaningful logs in the output console. I added the 'SIGABRT' one in the bottom of this bug anyway.
I have no specific logs in the device log. 

Please help me. 

### Steps to Reproduce

I did not manage to get the exact things producing this issue.
I tried to disable most of the things specific to release mode (AOT, triming) with no luck. I delete bin / obj folders between each attempt.

My release configuration :
```xml
<NoWarn>4014</NoWarn>
<JavaMaximumHeapSize>1G</JavaMaximumHeapSize>
<AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>
<AndroidSigningKeyStore>../misc/Keystore/JonathanAntoine/JonathanAntoine.keystore</AndroidSigningKeyStore>
<AndroidLinkMode>None</AndroidLinkMode>
<AndroidLinkSkip>
    TVST.Domain;TVST.DataAccess;TVST.Helpers;TVST.Core;Plugin.InAppBilling;Xamarin.Android.Google.BillingClient;
</AndroidLinkSkip>
<AndroidDexTool>d8</AndroidDexTool>
<AndroidLinkTool>r8</AndroidLinkTool>
<AndroidEnableMultiDex>true</AndroidEnableMultiDex>
<AndroidStoreUncompressedFileExtensions>.dll;</AndroidStoreUncompressedFileExtensions>
<EmbedAssembliesIntoApk>true</EmbedAssembliesIntoApk>
<RunAOTCompilation>false</RunAOTCompilation>
<EnableLLVM>false</EnableLLVM>
<AndroidEnableProfiledAot>false</AndroidEnableProfiledAot>
<MandroidI18n />
<AndroidCreatePackagePerAbi>false</AndroidCreatePackagePerAbi>
<AndroidUseAapt2>true</AndroidUseAapt2>
<BundleAssemblies>false</BundleAssemblies>
<RuntimeIdentifiers>android-arm;android-x86;android-x64</RuntimeIdentifiers>
<PublishTrimmed>false</PublishTrimmed>
<AndroidLinkResources>true</AndroidLinkResources>
<TrimMode>copyused</TrimMode>
```

I had to remove AppCenter because of crash at startup too, here is my existing list of packages :
```xml
<PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
<PackageReference Include="Microsoft.NETCore.Platforms" Version="7.0.4" />
<PackageReference Include="Microsoft.Win32.Primitives" Version="4.3.0" />
<PackageReference Include="Naxam.SimpleRatingBar.Droid" Version="0.1.5" />
<PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
<PackageReference Include="Plugin.CurrentActivity" Version="2.1.0.4" />
<PackageReference Include="Plugin.InAppBilling" Version="7.1.0" />
<PackageReference Include="sqlite-net-pcl" Version="1.8.116" />
<PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.7" />
<PackageReference Include="SQLitePCLRaw.core" Version="2.1.7" />
<PackageReference Include="SQLitePCLRaw.lib.e_sqlite3.android" Version="2.1.7" />
<PackageReference Include="SQLitePCLRaw.provider.e_sqlite3.android" Version="1.1.14" />
<PackageReference Include="Square.OkHttp3" Version="4.12.0.1" />
<PackageReference Include="Square.OkIO" Version="3.6.0.1" />
<PackageReference Include="Square.Picasso" Version="2.71828.0" />
<PackageReference Include="Xamarin.Android.Arch.Core.Common" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Core.Runtime" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Lifecycle.Common" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Lifecycle.LiveData" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Lifecycle.LiveData.Core" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Lifecycle.Runtime" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Arch.Lifecycle.ViewModel" Version="1.1.1.3" />
<PackageReference Include="Xamarin.Android.Support.Animated.Vector.Drawable" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Annotations" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.AsyncLayoutInflater" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Collections" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Compat" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.CoordinaterLayout" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Core.UI" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Core.Utils" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.CursorAdapter" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.CustomTabs" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.CustomView" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Design" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.DocumentFile" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.DrawerLayout" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Exif" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Fragment" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Interpolator" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Loader" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.LocalBroadcastManager" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Media.Compat" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Print" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.SlidingPaneLayout" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.SwipeRefreshLayout" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.Transition" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.v4" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.v7.AppCompat" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.v7.CardView" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.v7.RecyclerView" Version="28.0.0.3" />
    <PackageReference Include="Xamarin.Android.Support.Vector.Drawable" Version="28.0.0.3" ExcludeAssets="build"/>
<PackageReference Include="Xamarin.Android.Support.VersionedParcelable" Version="28.0.0.3" />
<PackageReference Include="Xamarin.Android.Support.ViewPager" Version="28.0.0.3" />
<PackageReference Include="Xamarin.AndroidX.Activity" Version="1.8.2" />
<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.2" />
<PackageReference Include="Xamarin.AndroidX.Annotation" Version="1.7.1" />
<PackageReference Include="Xamarin.AndroidX.AppCompat" Version="1.6.1.6" />
<PackageReference Include="Xamarin.AndroidX.AppCompat.AppCompatResources" Version="1.6.1.7" />
<PackageReference Include="Xamarin.AndroidX.AppCompat.Resources">
    <Version>1.1.0.2</Version>
</PackageReference>
<PackageReference Include="Xamarin.AndroidX.Arch.Core.Common" Version="2.2.0.6" />
<PackageReference Include="Xamarin.AndroidX.Arch.Core.Runtime" Version="2.2.0.6" />
<PackageReference Include="Xamarin.AndroidX.AsyncLayoutInflater" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.3.0.2" />
    <PackageReference Include="Xamarin.AndroidX.Collection" Version="1.3.0.2"/>-->
<PackageReference Include="Xamarin.AndroidX.CoordinatorLayout" Version="1.2.0.10" />
<PackageReference Include="Xamarin.AndroidX.Core" Version="1.12.0.3" />
<PackageReference Include="Xamarin.AndroidX.CursorAdapter" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.CustomView" Version="1.1.0.21" />
<PackageReference Include="Xamarin.AndroidX.DocumentFile" Version="1.0.1.22" />
<PackageReference Include="Xamarin.AndroidX.DrawerLayout" Version="1.2.0.6" />
<PackageReference Include="Xamarin.AndroidX.ExifInterface" Version="1.3.7" />
<PackageReference Include="Xamarin.AndroidX.Fragment" Version="1.6.2.1" />
<PackageReference Include="Xamarin.AndroidX.Interpolator" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.Legacy.Support.Core.UI" Version="1.0.0.23" />
<PackageReference Include="Xamarin.AndroidX.Legacy.Support.Core.Utils" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.Legacy.Support.V4" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.Common" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData.Core" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModelSavedState" Version="2.7.0" />
<PackageReference Include="Xamarin.AndroidX.Loader" Version="1.1.0.22" />
<PackageReference Include="Xamarin.AndroidX.LocalBroadcastManager" Version="1.1.0.10" />
<PackageReference Include="Xamarin.AndroidX.Media" Version="1.7.0" />
<PackageReference Include="Xamarin.AndroidX.Migration" Version="1.0.10" />
<PackageReference Include="Xamarin.AndroidX.MultiDex" Version="2.0.1.22" />
<PackageReference Include="Xamarin.AndroidX.Print" Version="1.0.0.22" />
<PackageReference Include="Xamarin.AndroidX.RecyclerView" Version="1.3.2.1" />
<PackageReference Include="Xamarin.AndroidX.SavedState" Version="1.2.1.6" />
<PackageReference Include="Xamarin.AndroidX.SlidingPaneLayout" Version="1.2.0.10" />
<PackageReference Include="Xamarin.AndroidX.SwipeRefreshLayout" Version="1.1.0.17" />
<PackageReference Include="Xamarin.AndroidX.Transition" Version="1.4.1.15" />
<PackageReference Include="Xamarin.AndroidX.VectorDrawable" Version="1.1.0.22" />
<PackageReference Include="Xamarin.AndroidX.VectorDrawable.Animated" Version="1.1.0.22" />
<PackageReference Include="Xamarin.AndroidX.VersionedParcelable" Version="1.1.1.22" />
<PackageReference Include="Xamarin.AndroidX.ViewPager" Version="1.0.0.22" />
<PackageReference Include="Xamarin.Build.Download" Version="0.11.4" />
<PackageReference Include="Xamarin.Firebase.Analytics" Version="121.3.0.4" />
<PackageReference Include="Xamarin.Google.Android.Material" Version="1.10.0.2" />
<PackageReference Include="Xamarin.Google.AutoValue.Annotations" Version="1.10.4.2" />
<PackageReference Include="Xamarin.GooglePlayServices.Ads.Lite" Version="122.3.0.2" />
<PackageReference Include="GoogleGson" Version="2.10.1.7" />
<PackageReference Include="Naxam.Flexbox.Droid" Version="0.3.1" />
<PackageReference Include="Xamarin.Kotlin.StdLib" Version="1.9.22" />
<PackageReference Include="Microsoft.Maui.Essentials" Version="8.0.0" />
```

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms, Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14 - Pixel Pro 6

### Did you find any workaround?

No :(

### Relevant log output

```shell
Time	Device Name	Type	PID	Tag	Message
02-04 20:43:57.335	Google Pixel 6 Pro	Error	6494	libc	Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 6494 (hanantoine.TVST), pid 6494 (hanantoine.TVST)
02-04 20:43:57.335	Google Pixel 6 Pro	Error	6494	hanantoine.TVST	* Assertion at /__w/1/s/src/mono/mono/metadata/object.c:4407, condition `is_ok (error)' not met, function:mono_unhandled_exception_internal, (null) assembly:Mono.Android.dll type:NoSuchMethodError member:(null)
02-04 20:43:57.331	Google Pixel 6 Pro	Info	6494	monodroid-timing	[1/5] Typemap.java_to_managed: end, total time; elapsed: 0:0::22705
02-04 20:43:57.331	Google Pixel 6 Pro	Debug	6494	monodroid-assembly	typemap: java type 'com/google/firebase/analytics/FirebaseAnalytics' corresponds to managed token id 33554443 (0x200000b)
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Running class .cctor for Firebase.Analytics.FirebaseAnalytics from 'Xamarin.GooglePlayServices.Measurement.Api.dll'
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Assembly Ref addref Xamarin.GooglePlayServices.Measurement.Api[0xe9eed690] -> System.Runtime[0xad545b90]: 41
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Assembly already loaded in the active ALC: 'System.Runtime'.
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Request to load System.Runtime in alc 0xea2203d0
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Loading reference 0 of Xamarin.GooglePlayServices.Measurement.Api.dll (default ALC), looking for System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
02-04 20:43:57.330	Google Pixel 6 Pro	Debug	6494	Mono	Requesting loading reference 0 (of 7) of Xamarin.GooglePlayServices.Measurement.Api.dll
02-04 20:43:57.329	Google Pixel 6 Pro	Debug	6494	Mono	Running class .cctor for Android.OS.BaseBundle from 'Mono.Android.dll'
02-04 20:43:57.329	Google Pixel 6 Pro	Debug	6494	Mono	Running class .cctor for Android.OS.Bundle from 'Mono.Android.dll'
```

**Comment by jonathanantoine** [comment]
@Eskissimo I am not sure it's related as I don't have LLVM enabled.

**Comment by steveisok** [comment]
@jonathanpeppers @jonpryor it's not clear to me if this is purely a runtime issue or something in the sdk. Can you take a look?

**Comment by jonathanantoine** [comment]
@jonathanpeppers I don't have LLVM enabled and not even profiled AOT. Please consider my release configuration from my issue and not the one from @Eskissimo.

I will try with these logs enabled ASAP @jonathanpeppers (not before 2 weeks unfortunately 😥)

@Eskissimo can you please create a separate issue for yours as its not the same one.

**Comment by jonathanantoine** [comment]
Hello,

I finally managed to retrieve the logs with the recommend settings, the issue seems to be this :

```
---> Java.Lang.IllegalStateException: You can consumeRestoredStateForKey only after super.onCreate of corresponding component
 --- End of managed Java.Lang.IllegalStateException stack trace ---
ava.lang.IllegalStateException: You can consumeRestoredStateForKey only after super.onCreate of corresponding component
	at androidx.savedstate.SavedStateRegistry.consumeRestoredStateForKey(SourceFile:72)
	at androidx.savedstate.Recreator.onStateChanged(SourceFile:34)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(SourceFile:322)
	at androidx.lifecycle.LifecycleRegistry.forwardPass(SourceFile:258)
	at androidx.lifecycle.LifecycleRegistry.sync(SourceFile:294)
	at androidx.lifecycle.LifecycleRegistry.moveToState(SourceFile:143)
	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(SourceFile:126)
	at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime_release(SourceFile:190)
	at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostCreated(SourceFile:113)
	at android.app.Activity.dispatchActivityPostCreated(Activity.java:1450)
	at android.app.Activity.performCreate(Activity.java:8778)
	at android.app.Activity.performCreate(Activity.java:8745)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1519)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3928)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4126)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:108)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:195)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:157)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:90)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2544)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8501)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
```


Full logs : 
[log.txt](https://github.com/dotnet/runtime/files/14628754/log.txt)

Can someone help me please ?

**Comment by jonathanpeppers** [comment]
Can you share what the `OnCreate` method looks like for: `com.jonathanantoine.TVST2/crc640e6af9aca67167cc.StartupView`?

Are you running some code before `base.OnCreate()`? That is what the Java error message indicates:

```
java.lang.IllegalStateException: You can consumeRestoredStateForKey only after super.onCreate of corresponding component
```

**Comment by sergiiPu** [comment]
try to use ProGuard to set rules , this is the only thing that helped me

**Comment by rpharon** [comment]
> try to use ProGuard to set rules , this is the only thing that helped me

can you show a screenshot?

**Comment by jonathanantoine** [comment]
Ouch, sorry for being so long to reply, here is the code of my view :

```csharp
protected override void OnCreate(Bundle savedInstanceState)
{
    base.OnCreate(savedInstanceState);

    // Check if we're running on Android 5.0 or higher
    if (Build.VERSION.SdkInt >= BuildVersionCodes.Lollipop)
    {
        // No transition
        RequestWindowFeature(WindowFeatures.ContentTransitions);
        Window.EnterTransition = new Android.Transitions.Fade(Android.Transitions.FadingMode.In);
    }

    RequestWindowFeature(WindowFeatures.NoTitle);

    SetContentView(Resource.Layout.StartupView);
    try
    {
        ServiceLocatorTvst.AnalyticsService.SendView("StartupView");
    }
    catch (Exception e)
    {
        e.WriteLineIfDebuggerAttached();
    }
         
    ViewModel = new StartupViewModel();
}
```

And here is the code that launch the activity : 

```csharp
var parameters = new Dictionary<string, string>
{
    {NavigationParameter.AutoConnect, autoConnect.ToString()}
};

var extras = string.Join("&", parameters.Select(k => k.Key + "=" + Uri.EscapeDataString(k.Value)));
var intent = new Intent(CrossCurrentActivity.Current.Activity, typeof(StartupView));
intent.PutExtra(ExtraKey, extras);
CrossCurrentActivity.Current.Activity.StartActivity(intent);

```

@sergiiPu which proguard rules to you use ?

**Comment by jonathanantoine** [comment]
Interesting update : it seems to work as soon as I use a classic `Activity` instead of an `AppCompatActivity` as a base class.

**Comment by jonathanpeppers** [comment]
If changing `AppCompatActivity` -> `Activity` would probably make the following code in your stack trace no longer called:
```
You can consumeRestoredStateForKey only after super.onCreate of corresponding component
	at androidx.savedstate.SavedStateRegistry.consumeRestoredStateForKey(SourceFile:72)
```

To be honest this seems like Google's bug, something going wrong in the AndroidX libraries. Similar issue here:

* https://stackoverflow.com/questions/58008221/android-crash-when-adding-livedata-observer

Java & Kotlin libraries can provide their own proguard rules, so perhaps they are missing an important one?

**Comment by jonathanantoine** [comment]
I have found out 2 solutions in between for those with the same issue: 
- remove all app compat old packages and fully migrate to AndroidX.
- if you still require app compat nugets : add the missing classes in your proguard rules
